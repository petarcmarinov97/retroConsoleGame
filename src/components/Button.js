import React, { Component } from 'react';
import { Button as CircleButton } from 'antd';

export default class Button extends Component {

    render() {
        const { size, style, shape, label, btnkey } = this.props
        const span = <span id="38" style={{ color: "black" }}>&#8635;</span>;

        return <>
            <CircleButton
                id={btnkey}
                btnkey={btnkey}
                style={style}
                size={size}
                type="primary"
                shape={shape}
            >
                {label && span}
            </CircleButton>
        </>
    }
}
