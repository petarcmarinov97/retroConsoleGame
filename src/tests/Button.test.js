import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";
import Button from "../components/Button";

const btn = {
    style: {
        "border": "1px solid #000",
        "borderColor": "transparent",
        "boxShadow": "0px 0px 2px 1px rgb(255 255 255 / 60%) inset",
        "visibility": "unset",
    },
    size: "large",
    shape: "circle",
    btnkey: 37,
}

describe('Button tests', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Button
                className={btn.className}
                style={btn.style}
                size={btn.size}
                shape={btn.shape}
                btnkey={btn.btnkey}
            />);
    })

    describe('Snapshots', () => {
        it("should render correctly", () => {
            expect(shallowToJson(wrapper)).toMatchSnapshot();
        })
    })
})
