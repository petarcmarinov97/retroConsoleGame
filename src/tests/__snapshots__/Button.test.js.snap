// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button tests Snapshots should render correctly 1`] = `
<Fragment>
  <Button
    btnkey={37}
    id={37}
    shape="circle"
    size="large"
    style={
      Object {
        "border": "1px solid #000",
        "borderColor": "transparent",
        "boxShadow": "0px 0px 2px 1px rgb(255 255 255 / 60%) inset",
        "visibility": "unset",
      }
    }
    type="primary"
  />
</Fragment>
`;
